### PR TITLE
client: remove deprecated client

### PR DESCRIPTION
**Pull request by bmwill**
This PR does the following:
* Remove the deprecated client
* Move the jsonrpc proto code into its own crate since its no longer used by anything in the codebase. We can use this crate as a place to put in some conversion tests between the views and the protos.

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  R prost 0.7.0 (direct third-party, crates.io)
    * (old features: default, prost-derive, std)

host packages:
  M getrandom 0.2.2 (transitive third-party, crates.io)
    * removed features: std
    * (unchanged features: [none])
  R prost-build 0.7.0 (direct third-party, crates.io)
    * (old features: [none])
  R bytes 1.0.1 (transitive third-party, crates.io)
    * (old features: [none])
  R either 1.6.1 (transitive third-party, crates.io)
    * (old features: [none])
  R fixedbitset 0.2.0 (transitive third-party, crates.io)
    * (old features: [none])
  R hashbrown 0.9.1 (transitive third-party, crates.io)
    * (old features: raw)
  R indexmap 1.6.2 (transitive third-party, crates.io)
    * (old features: [none])
  R itertools 0.9.0 (transitive third-party, crates.io)
    * (old features: default, use_std)
  R multimap 0.8.2 (transitive third-party, crates.io)
    * (old features: [none])
  R petgraph 0.5.1 (transitive third-party, crates.io)
    * (old features: [none])
  R ppv-lite86 0.2.10 (transitive third-party, crates.io)
    * (old features: simd, std)
  R prost 0.7.0 (transitive third-party, crates.io)
    * (old features: prost-derive)
  R prost-derive 0.7.0 (transitive third-party, crates.io)
    * (old features: [none])
  R prost-types 0.7.0 (transitive third-party, crates.io)
    * (old features: [none])
  R rand 0.8.3 (transitive third-party, crates.io)
    * (old features: alloc, default, getrandom, libc, rand_chacha, rand_hc, std, std_rng)
  R rand_chacha 0.3.0 (transitive third-party, crates.io)
    * (old features: std)
  R rand_core 0.6.2 (transitive third-party, crates.io)
    * (old features: alloc, getrandom, std)
  R remove_dir_all 0.5.3 (transitive third-party, crates.io)
    * (old features: [none])
  R tempfile 3.2.0 (transitive third-party, crates.io)
    * (old features: [none])
  R thiserror 1.0.24 (transitive third-party, crates.io)
    * (old features: [none])
  R which 4.0.2 (transitive third-party, crates.io) (other versions: 3.1.1)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  R prost 0.7.0 (direct third-party, crates.io)
    * (old features: default, prost-derive, std)

host packages:
  M getrandom 0.2.2 (transitive third-party, crates.io)
    * removed features: std
    * (unchanged features: [none])
  R prost-build 0.7.0 (direct third-party, crates.io)
    * (old features: [none])
  R bytes 1.0.1 (transitive third-party, crates.io)
    * (old features: [none])
  R either 1.6.1 (transitive third-party, crates.io)
    * (old features: [none])
  R fixedbitset 0.2.0 (transitive third-party, crates.io)
    * (old features: [none])
  R hashbrown 0.9.1 (transitive third-party, crates.io)
    * (old features: raw)
  R indexmap 1.6.2 (transitive third-party, crates.io)
    * (old features: [none])
  R itertools 0.9.0 (transitive third-party, crates.io)
    * (old features: default, use_std)
  R multimap 0.8.2 (transitive third-party, crates.io)
    * (old features: [none])
  R petgraph 0.5.1 (transitive third-party, crates.io)
    * (old features: [none])
  R ppv-lite86 0.2.10 (transitive third-party, crates.io)
    * (old features: simd, std)
  R prost 0.7.0 (transitive third-party, crates.io)
    * (old features: prost-derive)
  R prost-derive 0.7.0 (transitive third-party, crates.io)
    * (old features: [none])
  R prost-types 0.7.0 (transitive third-party, crates.io)
    * (old features: [none])
  R rand 0.8.3 (transitive third-party, crates.io)
    * (old features: alloc, default, getrandom, libc, rand_chacha, rand_hc, std, std_rng)
  R rand_chacha 0.3.0 (transitive third-party, crates.io)
    * (old features: std)
  R rand_core 0.6.2 (transitive third-party, crates.io)
    * (old features: alloc, getrandom, std)
  R remove_dir_all 0.5.3 (transitive third-party, crates.io)
    * (old features: [none])
  R tempfile 3.2.0 (transitive third-party, crates.io)
    * (old features: [none])
  R thiserror 1.0.24 (transitive third-party, crates.io)
    * (old features: [none])
  R which 4.0.2 (transitive third-party, crates.io) (other versions: 3.1.1)
    * (old features: [none])
```

File: json-rpc/types/src/views.rs [hunk 1]
```diff
@@ -1011,8 +1008,5 @@ mod tests {
                 }
             })
         );
-
-        let preburn: jsonrpc::PreburnWithMetadata = serde_json::from_value(value).unwrap();
-        assert_eq!(preburn.metadata, "");
```
Comment (xli) [inline json-rpc/types/src/views.rs:1016]: We need this to confirm the data rendered can be loaded back by protobuf structs.
We don't have other places cover this yet as we all moved to use same view structs.

Comment (bmwill) [reply json-rpc/types/src/views.rs:1016]: I didn't remove this test, if you look at the proto crate added in this commit, that test is moved there

File: json-rpc/types/proto/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            json!({
+                "preburn": {
+                    "amount": 1,
+                    "currency": "XUS"
+                }
+            })
+        );
+
+        let preburn: jsonrpc::PreburnWithMetadata = serde_json::from_value(value).unwrap();
+        assert_eq!(preburn.metadata, "");
```
Comment (bmwill) [inline json-rpc/types/proto/src/lib.rs:69]: @xli The test was moved here

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  R prost 0.7.0 (direct third-party, crates.io)
    * (old features: default, prost-derive, std)

host packages:
  M getrandom 0.2.2 (transitive third-party, crates.io)
    * removed features: std
    * (unchanged features: [none])
  R prost-build 0.7.0 (direct third-party, crates.io)
    * (old features: [none])
  R bytes 1.0.1 (transitive third-party, crates.io)
    * (old features: [none])
  R either 1.6.1 (transitive third-party, crates.io)
    * (old features: [none])
  R fixedbitset 0.2.0 (transitive third-party, crates.io)
    * (old features: [none])
  R hashbrown 0.9.1 (transitive third-party, crates.io)
    * (old features: raw)
  R indexmap 1.6.2 (transitive third-party, crates.io)
    * (old features: [none])
  R itertools 0.9.0 (transitive third-party, crates.io)
    * (old features: default, use_std)
  R multimap 0.8.2 (transitive third-party, crates.io)
    * (old features: [none])
  R petgraph 0.5.1 (transitive third-party, crates.io)
    * (old features: [none])
  R ppv-lite86 0.2.10 (transitive third-party, crates.io)
    * (old features: simd, std)
  R prost 0.7.0 (transitive third-party, crates.io)
    * (old features: prost-derive)
  R prost-derive 0.7.0 (transitive third-party, crates.io)
    * (old features: [none])
  R prost-types 0.7.0 (transitive third-party, crates.io)
    * (old features: [none])
  R rand 0.8.3 (transitive third-party, crates.io)
    * (old features: alloc, default, getrandom, libc, rand_chacha, rand_hc, std, std_rng)
  R rand_chacha 0.3.0 (transitive third-party, crates.io)
    * (old features: std)
  R rand_core 0.6.2 (transitive third-party, crates.io)
    * (old features: alloc, getrandom, std)
  R remove_dir_all 0.5.3 (transitive third-party, crates.io)
    * (old features: [none])
  R tempfile 3.2.0 (transitive third-party, crates.io)
    * (old features: [none])
  R thiserror 1.0.24 (transitive third-party, crates.io)
    * (old features: [none])
  R which 4.0.2 (transitive third-party, crates.io) (other versions: 3.1.1)
    * (old features: [none])
```

**Review by xli — APPROVED** [review]

**Pull request merged**